src/platforms/esp32/components/avm_builtins/network_driver.c

```diff
@@ -723,7 +744,6 @@ static void start_network(Context *ctx, term pid, term ref, term config)
         if ((err = esp_wifi_set_config(ESP_IF_WIFI_AP, ap_wifi_config)) != ESP_OK) {
             ESP_LOGE(TAG, "Error setting AP mode config %d", err);
             free(ap_wifi_config);
-            free(sta_wifi_config);
```

Comment:
I'm not sure how this causes a crash. Can you elaborate further?

Reply:
This pointer has already been free'd between lines 724 and 727.

Reply:
@bettio I understand the crash could also be avoided if sta_wifi_config was set to NULL after line 716 (old) 736 (new). If sta_wifi_config was not NULL, it is freed anyway in the block above.

Reply:
Here is from release-0.6:
https://github.com/atomvm/AtomVM/blob/eb406a82e15c53d77304d30ceabcfc67a2d5e593/src/platforms/esp32/components/avm_builtins/network_driver.c#L724C1-L738C6
```
    if (!IS_NULL_PTR(sta_wifi_config)) {
        if ((err = esp_wifi_set_config(ESP_IF_WIFI_STA, sta_wifi_config)) != ESP_OK) {
            ESP_LOGE(TAG, "Error setting STA mode config %d", err);
            free(sta_wifi_config);
            if (!IS_NULL_PTR(ap_wifi_config)) {
                free(ap_wifi_config);
            }
            term error = port_create_error_tuple(ctx, term_from_int(err));
            port_send_reply(ctx, pid, ref, error);
            return;
        } else {
            ESP_LOGI(TAG, "STA mode configured");
            free(sta_wifi_config);
        }
    }
```
Either path through this code will `free(sta_wifi_config)`. Freeing again below will causer a hard fault.

Reply:
> @bettio I understand the crash could also be avoided if sta_wifi_config was set to NULL after line 716 (old) 736 (new). If sta_wifi_config was not NULL, it is freed anyway in the block above.

This might be an easier way to avoid the bug, but I still am not a fan of freeing the same variable multiple times, it always makes me have to stop an re-examine the code the next time I’m working on a feature, because it looks suspicious.